src/output-tx.c

```diff
@@ -119,12 +120,24 @@ static TmEcode OutputTxLog(ThreadVars *tv, Packet *p, void *thread_data, PacketQ
 
     uint64_t total_txs = AppLayerParserGetTxCnt(p->proto, alproto, alstate);
     uint64_t tx_id = AppLayerParserGetTransactionLogId(f->alparser);
-    int tx_progress_done_value_ts =
-        AppLayerParserGetStateProgressCompletionStatus(p->proto, alproto,
-                                                       STREAM_TOSERVER);
-    int tx_progress_done_value_tc =
-        AppLayerParserGetStateProgressCompletionStatus(p->proto, alproto,
-                                                       STREAM_TOCLIENT);
+
+    int tx_progress_done_value_ts = 0;
+    int tx_progress_done_value_tc = 0;
+
+    /* we want to log TLS once the handshake is done */
+    if (alproto == ALPROTO_TLS) {
+        tx_progress_done_value_ts = TLS_HANDSHAKE_DONE;
```

Comment:
I think we should make this a configurable setting when we register the logger. Register a value for when the tx is 'done enough' to log. Make sense?